test/TPP/transform/transform-convolutions.mlir

```diff
@@ -285,3 +290,60 @@ transform.sequence failures(propagate) {
     %9 = transform.structured.interchange %8 { iterator_interchange = [0, 1, 4, 2, 3, 5] }
     transform.structured.map_to_brgemm %9
 }
+
+// -----
+
+#map = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
+func.func @walk(%1: tensor<1x56x56x256xf32>, %arg0: tensor<1x56x56x64xf32>, %arg1: tensor<1x1x64x256xf32>, %arg2: tensor<1x56x56x64xf32>, %arg3: tensor<1x1x64x256xf32>, %arg4: tensor<1x56x56x256xf32>) -> tensor<1x56x56x256xf32> {
+  // CHECK-DAG: %[[C0:.+]] = arith.constant 0 : index
+  // CHECK-DAG: %[[C1:.+]] = arith.constant 1 : index
+  // CHECK-DAG: %[[C2:.+]] = arith.constant 2 : index
+  // CHECK-DAG: %[[C8:.+]] = arith.constant 8 : index
+  // CHECK-DAG: %[[C56:.+]] = arith.constant 56 : index
+  // CHECK-DAG: %[[C32:.+]] = arith.constant 32 : index
+  %0 = tensor.empty() : tensor<1x56x56x256xf32>
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C8]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C56]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C56]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C32]] step %[[C1]] iter_args
+  // CHECK: %{{.*}} = scf.for %{{.*}} = %[[C0]] to %[[C2]] step %[[C1]] iter_args
+  // CHECK-COUNT-2: %{{.*}} = linalg.matmul ins(%{{.*}}, %{{.*}} : tensor<1x32xf32>, tensor<32x1xf32>) outs(%{{.*}} : tensor<1x1xf32>) -> tensor<1x1xf32>
+  // CHECK: %{{.*}} = linalg.generic {{.*}}, iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel"], library_call = "tpp.add"} ins(%{{.*}} : tensor<1x1x1x1x1xf32>) outs(%{{.*}} : tensor<1x1x1x1x1xf32>)
+  %2 = linalg.conv_2d_nhwc_hwcf {dilations = dense<1> : tensor<2xi64>, strides = dense<1> : tensor<2xi64>} ins(%arg0, %arg1 : tensor<1x56x56x64xf32>, tensor<1x1x64x256xf32>) outs(%0 : tensor<1x56x56x256xf32>) -> tensor<1x56x56x256xf32>
+  %3 = linalg.conv_2d_nhwc_hwcf {dilations = dense<1> : tensor<2xi64>, strides = dense<1> : tensor<2xi64>} ins(%arg2, %arg3 : tensor<1x56x56x64xf32>, tensor<1x1x64x256xf32>) outs(%1 : tensor<1x56x56x256xf32>) -> tensor<1x56x56x256xf32>
+  %4 = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%2 : tensor<1x56x56x256xf32>) outs(%3 : tensor<1x56x56x256xf32>) {
+    ^bb0(%in: f32, %out: f32):
+      %5 = arith.addf %in, %out : f32
+      linalg.yield %5 : f32
+  } -> tensor<1x56x56x256xf32>
+  return %4 : tensor<1x56x56x256xf32>
+}
+
+// X = linalg.conv_2d_nhwc_hwcf
+// Y = linalg.conv_2d_nhwc_hwcf
+// add(X, Y)
```

Comment:
If this is a mock resnet example, the pattern is, in its most simplified form:
```
conv -> relu -> conv -> relu -> add -> conv -> relu
          \---------------------^
```

Reply:
yes, it is a simplified version.